test/Core.Test/Utilities/CoreHelpersTests.cs

```diff
@@ -0,0 +1,146 @@
+        {
+            // Arrange
+            var item0 = Guid.NewGuid();
+            var item1 = Guid.NewGuid();
+
+            var ids = new[] { item0, item1 };
+
+            // Act
+            var dt = ids.ToGuidIdArrayTVP();
+
+            // Assert
+            Assert.Single(dt.Columns);
+            Assert.Equal("GuidId", dt.Columns[0].ColumnName);
+            Assert.Equal(2, dt.Rows.Count);
+            Assert.Equal(item0, dt.Rows[0][0]);
+            Assert.Equal(item1, dt.Rows[1][0]);
+        }
+
+        // TODO: Test the other ToArrayTVP Methods
+
+        [Theory]
+        [InlineData("12345&6789", "123456789")]
+        [InlineData("abcdef", "ABCDEF")]
+        [InlineData("1!@#$%&*()_+", "1")]
+        public void CleanCertificateThumbprint_Success(string input, string output)
+        {
+            // Arrange & Act
+            var sanitizedInput = CoreHelpers.CleanCertificateThumbprint(input);
+
+            // Assert
+            Assert.Equal(output, sanitizedInput);
+        }
+
+        // TODO: Add more tests
+        [Theory]
+        [InlineData("2020-12-30T11:49:12.0000000Z", 1609310952000L)]
+        public void ToEpocMilliseconds_Success(string dateString, long milliseconds)
+        {
+            // Arrange
+            var date = DateTime.Parse(dateString);
+
+            // Act & Assert
+            Assert.Equal(milliseconds, CoreHelpers.ToEpocMilliseconds(date));
+        }
+
+        [Theory]
+        [InlineData(1609310952000L, "2020-12-30T11:49:12.0000000Z")]
+        public void FromEpocMilliseconds(long milliseconds, string dateString)
+        {
+            // Arrange
+            var date = DateTime.Parse(dateString);
+
+            // Act & Assert
+            Assert.Equal(date, CoreHelpers.FromEpocMilliseconds(milliseconds));
+        }
+
+        [Fact]
+        public void SecureRandomString_Success()
+        {
+            // Arrange & Act
+            var @string = CoreHelpers.SecureRandomString(8);
+
+            // Assert
+            // TODO: Should probably add more Asserts down the line
+            Assert.Equal(8, @string.Length);
+        }
+
+        [Theory]
+        // [InlineData(long.MinValue, "8589934592 GB")]
+        // [InlineData(long.MinValue + 1, "Change")]
+        [InlineData(1000L, "1000 Bytes")]
+        [InlineData(1, "1 Bytes")]
+        [InlineData(-5L, "5 Bytes")]
+        [InlineData(long.MaxValue, "8589934592 GB")]
+        public void ReadableBytesSize_Success(long size, string readable)
+        {
+            // Act & Assert
+            Assert.Equal(readable, CoreHelpers.ReadableBytesSize(size));
+        }
+
+        [Fact]
+        public void CloneObject_Success()
+        {
+            var orignial = new { Message = "Message" };
+
+            var copy = CoreHelpers.CloneObject(orignial);
+
+            Assert.Equal(orignial.Message, copy.Message);
+        }
+
+        [Fact]
+        public void ExtendQuery_Success()
```

Comment:
I think It's worth quickly extending this to the following cases to develop a bit more of a contract

* uri with no parameters
* overriding a param value
* empty dictionary doesn't change the uri

Reply:
This is a very good idea, I will add all those and try to push it to it's limits a bit more.